configure.ml

```diff
@@ -1014,6 +1019,66 @@ let write_dbg_wrapper f =
 
 let _ = write_dbg_wrapper "dev/ocamldebug-coq"
 
+(** * Build .dir-locals.el *)
+
+module Sexp = struct
+  (* S-expressions *)
+  type t =
+    | Sym of string (* unquoted symbol [sym] *)
+    | String of string (* string litteral ["foo"] *)
+    | Cons of t * t (* cons cell [(a . b)] *)
+    | List of t list (* list [(x1 ... xn)] *)
+
+  let rec pr o = function
+    | Sym s -> fprintf o "%s" s
+    | String s -> fprintf o "\"%s\"" s
+    | Cons (a,b) -> fprintf o "("; pr o a; fprintf o " . "; pr o b; fprintf o ")"
+    | List l ->
+      let rec fold = function
+        | [] -> ()
+        | [x] -> pr o x
+        | x::l -> pr o x; fprintf o " "; fold l
+      in
+      fprintf o "(";
+      fold l;
+      fprintf o ")"
+
+  let nil = Sym "nil"
+  let t = Sym "t"
+
+  let set_string sym v = Cons (Sym sym, String v)
+  let string_list l = List (List.map (fun x -> String x) l)
+
+  (* Make the dir-locals sexp for a repository located at [dir] (without ending slash) *)
+  let make_locals dir =
+    let allmodes =
+      let tags = set_string "tags-file-name" (dir^"/TAGS") in
+      let camldebug = set_string "camldebug-command-name" (dir^"/dev/ocamldebug-coq") in
+      let compile = set_string "compile-command" ("make -C "^dir) in
+      let compile_search = Cons (Sym "compilation-search-path", Cons (String dir, nil)) in
+      Cons (nil, List [tags;camldebug;compile;compile_search])
+    in
+    let coqmode =
+      let coqprog = set_string "coq-prog-name" (dir^"/bin/coqtop") in
+      Cons (Sym "coq-mode", List [coqprog])
+    in
+    List [allmodes;coqmode]
+end
+
+let write_dir_locals f =
+  safe_remove f;
+  let o = open_out f in
+  let pr s = fprintf o s in
+  pr ";; EMACS CONFIGURATION FOR COQ DEVELOPPERS\n";
+  pr ";; This configuration will be executed for each opened file under coq root directory.\n";
+  pr ";; DO NOT EDIT THIS FILE: automatically generated by ../configure\n";
```

Comment:
Do you mean `./configure` rather than `../configure`?

Reply:
>Do you mean ./configure rather than ../configure?

oops

Reply:
>Do you mean ./configure rather than ../configure?

Ended up using just `configure`.